src/odemis/driver/smaract.py

```diff
@@ -1255,8 +1255,6 @@ def __init__(self, name, role, locator, axes, ref_on_init=False, linear_speed=0.
         else:
             if ref_on_init:
                 self.reference().result()
```

Comment:
Referencing shouldn't be blocking.
If referencing fails, I'm fine with the current implementation about changing the state directly inside the referencing function. So this means you don't need to put a callback. Just call `self.reference()` and add a comment that it's going to run in the background.